File: week1/pset1/mario/less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;
    // Ensure that the user type a positive value between 1 and 8, inclusive
    do
    {
        height = get_int("Height: ");
    }
    while (height <= 0 || height > 8);

    int n = height;

    for (int i = 0; i < height; i++)
    {
        // Print the spaces
        for (int s = n - 1; s > 0; s--)
        {
            printf(" ");
        }
        // Print the hashes
        for (int j = i + 1; j > 0; j--)
        {
            printf("#");
        }
        // The number of spaces decreases at each row
        n --;
        // In the end of the internal loop, jump to the next line
        printf("\n");
    }
}

File: week6/pset6/sentimental-mario/less/mario.py
from cs50 import get_int
# Ensure that the user type a positive value between 1 and 8, inclusive
while True:
    height = get_int("Height: ")
    if height > 0 and height <= 8:
        break
# A variable we'll play with
n = height
# Iterates through the half-pyramid's height
for i in range(height):
    # Print the spaces, for identation
    print(" " * (n - 1), end="")
    # At each iteration, decrease the number of spaces printed
    n -= 1
    # Print ones pyramid's row
    print("#" * (i + 1), end="")
    # Jump to the next line
    print()

